packages/belos/src/BelosBlockCGSolMgr.hpp

```diff
@@ -828,6 +838,8 @@ BlockCGSolMgr<ScalarType,MV,OP,true>::getValidParameters() const
       "The string to use as a prefix for the timer labels.");
     pl->set("Orthogonalization", static_cast<const char *>(orthoType_default_),
       "The type of orthogonalization to use: DGKS, ICGS, or IMGS.");
+    pl->set("Assert Positive Definiteness",static_cast<bool>(assertPositiveDefiniteness_default_),
```

Comment:
It's a `bool`; you don't need the `static_cast`.

Reply:
I think you're right -- any idea why, in that block of code, there are several static casts of things to ints and bools?  Even for the static casts to const char*, those variables that are being casted seem to already be of those types.  Are any of these static casts here actually necessary?

Reply:
@DABH The casts are probably not necessary.  Belos is nearly 15 years old -- it's possible that some compilers we had to support back that had trouble deducing some of those types.  It's also likely that later developers went overboard with the casts.  I'll be happy to approve PRs that purge these; I just don't have time at the moment to clean this up.

Reply:
btw I'm also not convinced that ParameterList needs overloads for `const char value[]` and `char value[]`; the former should suffice.  [This example](https://godbolt.org/z/hXjTcB) compiles even with C++98.